src/test/java/org/jabref/logic/exporter/DocbookExporterTest.java

```diff
@@ -0,0 +1,86 @@
+package org.jabref.logic.exporter;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.logic.layout.LayoutFormatterPreferences;
+import org.jabref.logic.xmp.XmpPreferences;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junitpioneer.jupiter.TempDirectory;
+import org.mockito.Answers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(TempDirectory.class)
+public class DocbookExporterTest {
+
+    public BibDatabaseContext databaseContext;
+    public Charset charset;
+
+    private Exporter exportFormat;
+
+    @BeforeEach
+    public void setUp() {
+        Map<String, TemplateExporter> customFormats = new HashMap<>();
+        LayoutFormatterPreferences layoutPreferences = mock(LayoutFormatterPreferences.class, Answers.RETURNS_DEEP_STUBS);
+        SavePreferences savePreferences = mock(SavePreferences.class);
+        XmpPreferences xmpPreferences = mock(XmpPreferences.class);
+        ExporterFactory exporterFactory = ExporterFactory.create(customFormats, layoutPreferences, savePreferences, xmpPreferences);
+
+        exportFormat = exporterFactory.getExporterByName("docbook").get();
+
+        databaseContext = new BibDatabaseContext();
+        charset = StandardCharsets.UTF_8;
```

Comment:
You can move `databaseContext` and `charset` initialization into the field, so the `setUp` method is less complex.